mypy/checkexpr.py

```diff
@@ -3199,7 +3199,7 @@ def analyze_super(self, e: SuperExpr, is_lvalue: bool) -> Type:
                         return AnyType(TypeOfAny.from_error)
                     declared_self = args[0].variable.type or fill_typevars(e.info)
                     return analyze_member_access(name=e.name,
-                                                 typ=fill_typevars(e.info),
+                                                 typ=declared_self,
```

Comment:
I'm not sure about this. This looks like it might look up the signature from the current class, not the base class?

Reply:
I think that's what `override_info` is for below, which makes the checkmember machinery use that instead of `typ` as the context to look up members.

Actually I think `override_info` could be removed, and instead just pass the base type as `typ`.  I'll try that out.

Reply:
I looked into this in more detail, the `override_info` is necessary to handle things like generics.  If we just pass the base type as `typ` it throws away too much information.